ui/raidboss/raidemulator.js

```diff
@@ -0,0 +1,832 @@
+      ],
+      NIN: [
+        2240, 2241, 2242, 2245, 2246, 2247, 2248, 2253, 2254, 2255, 2256, 2257, 2258, 2259,
+        2260, 2261, 2262, 2263, 2264, 2265, 2266, 2267, 2268, 2269, 2270, 2271, 2272, 3563,
+        3564, 3565, 3566, 3567, 7401, 7402, 7403,
+      ],
+      SAM: [
+        7477, 7478, 7479, 7480, 7481, 7482, 7483, 7484, 7485, 7486, 7487, 7488, 7489, 7490,
+        7491, 7492, 7493, 7494, 7495, 7496, 7497, 7498, 7499, 7500, 7501, 7502, 7867,
+      ],
+      BRD: [
+        97, 98, 100, 101, 103, 106, 107, 110, 112, 113, 114, 115, 116, 117, 118, 3558, 3559,
+        3560, 3561, 3562, 7404, 7405, 7406, 7407, 7408, 7409,
+      ],
+      MCH: [
+        2864, 2865, 2866, 2867, 2868, 2870, 2872, 2873, 2874, 2875, 2876, 2878, 2879, 2880,
+        2881, 2885, 2887, 2888, 2890, 3487, 7410, 7411, 7412, 7413, 7414, 7415, 7418, 9015, 9372,
+      ],
+      BLM: [
+        141, 142, 144, 145, 146, 147, 149, 152, 153, 154, 155, 156, 157, 158, 159, 162,
+        3573, 3574, 3575, 3576, 3577, 7419, 7420, 7421, 7422, 7447,
+      ],
+      SMN: [
+        180, 181, 182, 184, 3578, 3579, 3580, 3581, 3582, 7423, 7424, 7425, 7426, 7427, 7429,
+      ],
+      RDM: [
+        7503, 7504, 7505, 7506, 7507, 7508, 7509, 7510, 7411, 7512, 7513, 7514, 7515, 7516,
+        7517, 7518, 7519, 7520, 7521, 7522, 7523, 7524, 7525, 7526, 7527, 7528, 7529, 7530,
+      ],
+    };
+    let players = this.players = {};
+    let regex = Regexes.Parse(/ 15:(\y{ObjectId}):(\y{Name}):(\y{AbilityCode}):/);
+
+    for (let i = 0; i < logs.length; i++) {
+      let matches = logs[i].match(regex);
+      if (matches != null) {
+        let ObjId = new RegExp(matches[1]);
+        if (!ObjId.test(Object.keys(players))) {
+          let playerId = matches[1];
+          let job;
+          let role;
+          let ability = parseInt(matches[3], 16);
+          for (let m in moves) {
+            for (let a = 0; a < moves[m].length; a++) {
+              if (ability == moves[m][a]) {
+                job = m;
+                break;
+              }
+            }
+            if (job)
+              break;
+          }
+          if (job == 'PLD' || job == 'WAR' || job == 'DRK')
+            role = 'tank';
+          if (job == 'WHM' || job == 'SCH' || job == 'AST')
+            role = 'healer';
+          if (job == 'MNK' || job == 'DRG' || job == 'NIN' || job == 'SAM' || job == 'BRD' || job == 'MCH' || job == 'BLM' || job == 'SMN' || job == 'RDM')
+            role = 'dps';
+          if (role && job) {
+            Object.assign(players, {
+              [playerId]: {
+                name: matches[2],
+                role: role,
+                job: job,
+                id: playerId,
+              },
+            });
+          }
+        }
+      }
+
+      // This can speed up this function a lot!
+      // Add logic to differ between 4 and 8 player dungeons
+      // TODO: Find a better Way of doing this

```

Comment:
This is an oddball thought, but when walking through all the logs to figure out where the fights are, you could figure out all the people that were hit by protect and use that to figure out party members in the zone before you even got to this point.

Reply:
Maybe the SearchPlayers function could be integrated into the LogCollector to prevent iterating over the same data again and again. I will try that!